src/AltinnCore/Runtime/RestControllers/ProcessController.cs

```diff
@@ -41,29 +66,445 @@ public class ProcessController : ControllerBase
+            }
+        }
+
+        private string CheckOrGetValidNextElement(string currentElementId, string proposedElementId, out ActionResult nextElementError)
+        {
+            nextElementError = null;
+            string nextElement = null;
+
+            List<string> possibleNextElements = ProcessKeeper.NextElements(currentElementId);
+            
+            if (!string.IsNullOrEmpty(proposedElementId) && possibleNextElements.Contains(proposedElementId))
+            {
+                nextElement = proposedElementId;
+            }
+            else if (possibleNextElements.Count == 1)
+            {
+                nextElement = possibleNextElements.First();
+            }
+            else if (!string.IsNullOrEmpty(proposedElementId))
+            {
+                nextElementError = Conflict($"Process element id '{proposedElementId}' is not found in app's process model (bpmn)");
+            }
+            else if (possibleNextElements.Count > 1)
+            {
+                nextElementError = Conflict($"There are more than one outgoing sequence flow, please select one '{possibleNextElements}'");
+            }
+
+            return nextElement;
+        }
+
+        /// <summary>
+        ///  Todo: Fix validation code of the instance's currentTask.
+        /// </summary>        
+        /// <returns>try if validation is OK, false otherwise</returns>
+        private bool InstanceIsValid(Instance instance)
+        {
+            return true;            
+        }
+
+        private List<InstanceEvent> ChangeProcessState(Instance instance, string nextElementId)
+        {
+            List<InstanceEvent> events = new List<InstanceEvent>();
+
+            ProcessState currentState = instance.Process;
+
+            string previousElementId = currentState.CurrentTask?.ElementId;
+
+            ElementInfo nextElementInfo = ProcessKeeper.GetElementInfo(nextElementId);
+
+            DateTime now = DateTime.UtcNow;
+            int flow = 1;
+
+            if (IsStartEvent(previousElementId))
+            {
+                flow = 1;
+            }
+
+            if (IsTask(previousElementId))
+            {
+                if (currentState.CurrentTask != null && currentState.CurrentTask.Flow.HasValue)
+                {
+                    flow = currentState.CurrentTask.Flow.Value;
+                }
+
+                events.Add(GenerateProcessChangeEvent("process:EndTask", instance, null, now));
+            }
+
+            if (IsEndEvent(nextElementId))
+            {
+                currentState.CurrentTask = null;
+                currentState.Ended = now;
+                currentState.EndEvent = nextElementId;
+
+                events.Add(GenerateProcessChangeEvent("process:EndEvent", instance, previousElementId, now));
+            }
+            else if (IsTask(nextElementId))
+            {
+                currentState.CurrentTask = new ProcessElementInfo
```

Comment:
shouldn't this be set as instance.process after this update?

Reply:
It's a referenced variable. I can add a extra logical  assignement at the end.